### ISSUE TITLE
Storm operators should validate form / prop types exist in the model

### ISSUE THREAD

**Issue by therealsilence** [vertexproject/synapse#299]
Storm operators (at minimum, addnode()) do not seem to validate that forms (and possibly properties?) exist before allowing node creation:
```
cli> ask addnode(inet:fqdn:hurrderp,umwut)
inet:fqdn:hurrderp = umwut
(1 results)

cli> ask --props inet:fqdn:hurrderp=umwut
inet:fqdn:hurrderp = umwut
(1 results)
```

**Comment by vEpiphyte** [comment]
Need to repro this with a cortex that has enforce enaled on it.

**Comment by vEpiphyte** [comment]
Reproduced this on current master cf55d1427b286e75cfda7c41e87bfcfcd7e5cdb0 using the following config:
```
{
  "vars": {
    "cortex_url": "ram://"
  },
  "ctors": [
    [
      "core",
      "ctor://synapse.cortex.openurl(cortex_url)",
      {"config": "coreopts"}
    ]
  ],
  "configs": {
    "coreopts": {
      "enforce": true
    }
  },
  "share": [
    [
      "core",
      {}
    ]
  ],
  "listen": [
    "tcp://127.0.0.1:47779"
  ]
}
```

Validated enforce it set

```
/some/path/synapse$ python
>>> import synapse.telepath as st
>>> p = st.openurl('tcp://127.0.0.1:47779/core')
>>> p.getConfOpt('enforce')
1
>>> exit()
```
the addnode() operator and edit syntax both create the nodes highlighted in the report

```
cli> ask --raw addnode(inet:fqdn:hurrderp, omgwut)
[
  [
    "3e1398c77846f76d2e3e656e76618363",
    {
      ".new": true,
      "inet:fqdn:hurrderp": "omgwut",
      "tufo:form": "inet:fqdn:hurrderp"
    }
  ]
]
(1 results)
cli> ask --raw inet:fqdn:hurrderp
[
  [
    "3e1398c77846f76d2e3e656e76618363",
    {
      "inet:fqdn:hurrderp": "omgwut",
      "tufo:form": "inet:fqdn:hurrderp"
    }
  ],
  [
    "2018d0fa942e347ed9c56dd8f6e5f96b",
    {
      "inet:fqdn:hurrderp": "umwut",
      "tufo:form": "inet:fqdn:hurrderp"
    }
  ]
]
(2 results)
cli> ask --raw [ inet:fqdn:hurrderp="this is bad" ]
[
  [
    "688e9f9c1ea0f0086fb8381b2c88fbc9",
    {
      ".new": true,
      "inet:fqdn:hurrderp": "this is bad",
      "tufo:form": "inet:fqdn:hurrderp"
    }
  ]
]
(1 results)

```

**Comment by e2-ibm** [comment]
Possibly related? Currently able to create an inet:srv4 which is a type and not a tufo:form

```
cli> ask --raw [ inet:srv4=8.8.8.8:80 ]
[
  [
    "6711989d1df28786cfb089a1957e930b",
    {
      ".new": true,
      "inet:srv4": "8.8.8.8:80",
      "tufo:form": "inet:srv4"
    }
  ]
]
(1 results)
```

visi [6 days ago] 
Actually, that was created extra-model.  that's a default we meant to change in enforce mode.  basically, the original idea ( which we now see is probably not the way to go ) is when a cortex is in model enforcement mode, if you try to create something that is un-modeled, but doesn't *conflict* with the current model definitions it allows it.  We'll be changing that, it just got back burnered for a bit.  i'll create an issue so we dont forget it again.


visi [6 days ago] 
( you can tell because the inet:srv4 data is still a string.  if type normalization had been applied, it would be a pure integer )

**Comment by vEpiphyte** [comment]
@e2-ibm I just realized I missed testing your extra-model condition there.

**Comment by vEpiphyte** [comment]
@e2-ibm The fix in #381 does prevent that node to be created when the cortex is in enforce mode.
